src/ClientContext.tsx

```diff
@@ -299,61 +249,141 @@ export const ClientProvider: FC<Props> = ({ children }) => {
     if (!widget) loadChannel?.postMessage({});
   }, []);
 
+  const [alreadyOpenedErr, setAlreadyOpenedErr] = useState<Error | undefined>(
+    undefined
+  );
   useEventTarget(
     loadChannel,
     "message",
     useCallback(() => {
-      client?.stopClient();
-
-      setState((prev) => ({
-        ...prev,
-        error: translatedError(
-          "This application has been opened in another tab.",
-          t
-        ),
-      }));
-    }, [client, setState, t])
+      initClientState?.client.stopClient();
+      setAlreadyOpenedErr(
+        translatedError("This application has been opened in another tab.", t)
+      );
+    }, [initClientState?.client, setAlreadyOpenedErr, t])
   );
 
-  const context = useMemo<ClientState>(
-    () => ({
-      loading,
-      isAuthenticated,
-      isPasswordlessUser,
-      client,
-      changePassword,
-      logout,
-      userName,
-      setClient,
-      error: undefined,
-    }),
-    [
-      loading,
-      isAuthenticated,
-      isPasswordlessUser,
-      client,
-      changePassword,
-      logout,
-      userName,
-      setClient,
-    ]
-  );
+  const [state, setState] = useState<ClientState | undefined>(undefined);
+  React.useEffect(() => {
```

Comment:
This combination of `useState` + `useEffect` could be replaced with a single `useMemo` hook to achieve the same thing with fewer renders

Reply:
My poor React skills, fixed 🙂 It looks much better with `useMemo()`, thanks!